cmd/ci-secret-bootstrap/main.go

```diff
@@ -0,0 +1,182 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/ghodss/yaml"
+	"github.com/sirupsen/logrus"
+	"io/ioutil"
+	coreapi "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	coreclientset "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"os"
+	"strings"
+
+	"github.com/openshift/ci-tools/pkg/bitwarden"
+	"github.com/openshift/ci-tools/pkg/util"
+)
+
+type options struct {
+	dryRun         bool
+	kubeConfigPath string
+	configPath     string
+	bwUser         string
+	bwPasswordPath string
+
+	bwPassword  string
+	kubeConfigs map[string]rest.Config
+	config      *Config
+}
+
+func parseOptions() options {
+	var o options
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.BoolVar(&o.dryRun, "dry-run", true, "Whether to actually create the secrets with oc command")
+	fs.StringVar(&o.kubeConfigPath, "kubeconfig", "", "Path to the kubeconfig file to use for CLI requests.")
+	fs.StringVar(&o.configPath, "config", "", "Path to the config file to use for this tool.")
+	fs.StringVar(&o.bwUser, "bw-user", "", "Username to access BitWarden.")
+	fs.StringVar(&o.bwPasswordPath, "bw-password-path", "", "Path to a password file to access BitWarden.")
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		logrus.WithError(err).Errorf("cannot parse args: '%s'", os.Args[1:])
+	}
+	return o
+}
+
+func (o *options) validateOptions() error {
+	if o.bwUser == "" {
+		return fmt.Errorf("the mandatory arg bw-user is empty")
+	}
+	if o.bwPasswordPath == "" {
+		return fmt.Errorf("the mandatory arg bw-password-path is empty")
+	}
+	if o.configPath == "" {
+		return fmt.Errorf("the mandatory arg config is empty")
+	}
+	return nil
+}
+
+func (o *options) completeOptions() error {
+	bytes, err := ioutil.ReadFile(o.bwPasswordPath)
+	if err != nil {
+		return err
+	}
+	o.bwPassword = strings.TrimSpace(string(bytes))
+
+	kubeConfigs, _, err := util.LoadKubeConfigs(o.kubeConfigPath)
+	if err != nil {
+		return err
+	}
+	o.kubeConfigs = kubeConfigs
+
+	config, err := loadConfig(o.configPath)
+	if err != nil {
+		return err
+	}
+	o.config = config
+
+	return validateConfig(kubeConfigs, config)
+}
+
+type Config struct {
+	Secrets []SecretConfig `json:"secrets"`
+}
+
+type BWContext struct {
+	BWItem     string `json:"bwItem"`
+	Field      string `json:"field"`
+	Attachment string `json:"attachment"`
```

Comment:
`Field` and `Attachment` are mutually exclusive, right? So these should be omitempty, and we should validate we have exactly non-empty one.

Reply:
Validation has listed in TODO.
https://github.com/openshift/ci-tools/blob/abf6fe6b64a51e1e2f3157f5513b08585ee500d4/cmd/ci-secret-bootstrap/main.go#L117-L119

Is `omitempty` only for serialization?
I think we do only deserialization here.
I am adding it anyway just in case.